### ISSUE TITLE
robovizDraw: java.nio.BufferOverflowException

### ISSUE THREAD

**Issue by ItsBean** [magmaOffenburg/magmaRelease#1]
```java
Consumer<RoboVizDraw> draw = roboVizDraw -> {
            Vector3D[] futureBallPosition = getWorldModel().getBall().getFuturePositions(100);
            int totalPositions = futureBallPosition.length;
            // draw the future ball position
            for (int i = 0; i < totalPositions; i+=20) {
                Vector3D future_ballPos = futureBallPosition[i];
                float factor = (float) i / totalPositions;
                // adjust the color according to the index, higher the index lighter the color
                int red = Math.max(0, (int) (255 - 255 * factor));
                int green = Math.max(0, (int) (215 - 215 * factor));
                int blue = (int) (255 * factor);
                roboVizDraw.drawPoint(null, future_ballPos, 2f, new Color(red, green, blue));
                
            }
            //draw summarization above our robot.
            Vector3D robotPos = getWorldModel().getThisPlayer().getPosition();
            Vector3D robotPos_head = robotPos.add(new Vector3D(0, 0, 0.9));
            // purple color:
            Color color = new Color(128, 0, 128);
            roboVizDraw.drawAnnotation(null, gpt4_summarization, robotPos_head, color);
            // print ballPos with red
            roboVizDraw.drawPoint(null, estimatedBallPoint, 10f, Color.RED);
        };
        getThoughtModel().getRoboVizDraw().asSingleDrawing(null, draw);
```

the bug is :
```
Crash!!! java.nio.BufferOverflowException
java.nio.BufferOverflowException
	at java.nio.Buffer.nextPutIndex(Buffer.java:525)
	at java.nio.HeapByteBuffer.put(HeapByteBuffer.java:173)
	at magma.util.roboviz.RoboVizBufferUtil.writeStringToBuffer(RoboVizBufferUtil.java:51)
	at magma.util.roboviz.RoboVizBufferUtil.newAnnotation(RoboVizBufferUtil.java:234)
	at magma.util.roboviz.RoboVizDraw.drawAnnotation(RoboVizDraw.java:255)
	at magma.util.roboviz.RoboVizDraw.drawAnnotation(RoboVizDraw.java:245)
	at magma.agent.decision.behavior.complex.misc.Attack.lambda$decideNextBasicBehaviors$0(Attack.java:95)
	at magma.util.roboviz.RoboVizDraw.asSingleDrawing(RoboVizDraw.java:371)
	at magma.agent.decision.behavior.complex.misc.Attack.decideNextBasicBehaviors(Attack.java:99)
	at hso.autonomy.agent.decision.behavior.complex.ComplexBehavior.perform(ComplexBehavior.java:58)
	at hso.autonomy.agent.decision.decisionmaker.impl.DecisionMakerBase.decide(DecisionMakerBase.java:104)
	at magma.agent.decision.decisionmaker.impl.SoccerGoalieDecisionMaker.decide(SoccerGoalieDecisionMaker.java:41)
	at hso.autonomy.agent.agentruntime.AgentRuntime.onDecide(AgentRuntime.java:134)
	at hso.autonomy.agent.agentruntime.AgentRuntime.update(AgentRuntime.java:91)
	at magma.agent.agentruntime.RoboCupAgentRuntime.update(RoboCupAgentRuntime.java:106)
	at hso.autonomy.agent.agentruntime.AgentRuntime.startClient(AgentRuntime.java:161)
	at magma.robots.RoboCupClient.main(RoboCupClient.java:42)
```
The string buffer seems to accumulate, so how do I flush the buffer every frame? Or how should I fix this?

**Comment by ItsBean** [comment]
Well, I found the mistake myself. The string `'decisionToDisplay: Go (0.47, 2.04; 66.20°): stay close, block shots, avoid penalty circle.'` contains the special character '°', which is not a standard ASCII character and it requires two bytes to store. Therefore, I suggest increasing the buffer size in the function `public static byte[] newAnnotation(String text, float[] pos, Color color, String set)`.

**Comment by hannesbraun** [comment]
Yes, that would fix the problem. I'll see if I've got the time to do that in the next weeks... As the competition in Bordeaux is getting closer, I don't have a lot of time right now.

**Comment by hannesbraun** [comment]
I just fixed this with 98aedca3b37c6ea38b2c762118c589e70c698758. Strings will now be encoded explicitly using UTF-8. I also updated RoboViz to support decoding UTF-8 strings sent through RoboViz Draw. (Otherwise, you might still see gibberish on your screen.)

Please let me know if this works for you too.

**Comment by ItsBean** [comment]
thank you!!
